### PR TITLE
Center the canvas on a newly placed bundle.

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2578/
Test FAILed.

**Comment by makyo** [comment]
# QA

```
juju deploy local:juju-gui --repository=.
juju set juju-gui juju-gui-source="https://github.com/makyo/juju-gui.git center-bundle"
juju expose juju-gui
```

Drag and drop a bundle and ensure the canvas centers on it when the changeset has been fully ingested.  Repeat for click to add to canvas.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2579/
Test PASSed.

File: test/test_bundle_importer.js [hunk 1]
```diff
@@ -299,9 +299,14 @@ describe('Bundle Importer', function() {
   });
 
   describe('Changeset execution', function() {
-    it('Sets up the correct environment (Integration)', function() {
+    it('Sets up the correct environment (Integration)', function(done) {
       var data = utils.loadFixture(
           'data/wordpress-bundle-recordset.json', true);
+      bundleImporter.db.after('bundleImportComplete', function() {
```
Comment (hatched) [inline test/test_bundle_importer.js]: While unlikely it's possible that this `done()` is called before the assertions below. I'd recommend moving them into this callback.

**Comment by hatched** [comment]
:+1: QA OK

Notes:
This snaps the user to the bundle after it's deployed so unfortunately it still doesn't look like anything is happening until it's completely done. At which point it snaps to the hull position.

I was hoping that this would position the canvas over the deploying service icons but I understand that without building in some easing this is jarring. What do you think about moving the canvas to the first deployed service from the bundle?

**Comment by makyo** [comment]
@hatched we've discussed the fact that I disagree with this approach.  I think you offer an interesting alternative, though.  Even so, I'd like for this to land more than I'd like to have further discussion on the topic.  Can we defer?

**Comment by hatched** [comment]
Yep, we can!

**Comment by kadams54** [comment]
:+1: QA is OK.

**Comment by makyo** [comment]
Awesome, thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**